src/vm/database/marf.rs

```diff
@@ -87,54 +102,97 @@ pub fn sqlite_marf(path_str: &str, miner_tip: Option<&BlockHeaderHash>) -> Resul
         None => TrieFileStorage::block_sentinel()
     };
 
-    Ok( MarfedKV { chain_tip, marf, side_store } )
+    Ok( MarfedKV { marf: Some(marf), chain_tip, side_store } )
 }
 
-impl <S> MarfedKV <S> where S: KeyValueStorage {
+impl MarfedKV {
+    pub fn as_clarity_db<'a>(&'a mut self) -> ClarityDatabase<'a> {
+        ClarityDatabase::new(self)
+    }
+
+    pub fn as_analysis_db<'a>(&'a mut self) -> AnalysisDatabase<'a> {
+        AnalysisDatabase::new(self)
+    }
+
+    /// begin, commit, rollback a save point identified by key
+    ///    this is used to clean up any data from aborted blocks
+    ///     (NOT aborted transactions that is handled by the clarity vm directly).
+    /// The block header hash is used for identifying savepoints.
+    ///     this _cannot_ be used to rollback to arbitrary prior block hash, because that
+    ///     blockhash would already have committed and no longer exist in the save point stack.
+    /// this is a "lower-level" rollback than the roll backs performed in
+    ///   ClarityDatabase or AnalysisDatabase -- this is done at the backing store level.
+
     pub fn begin(&mut self, current: &BlockHeaderHash, next: &BlockHeaderHash) {
-        self.marf.begin(current, next)
+        let marf = self.marf.as_mut().unwrap();
+        marf.begin(current, next)
             .expect(&format!("ERROR: Failed to begin new MARF block {} - {})", current.to_hex(), next.to_hex()));
-        self.chain_tip = self.marf.get_open_chain_tip()
+        self.chain_tip = marf.get_open_chain_tip()
             .expect("ERROR: Failed to get open MARF")
             .clone();
         self.side_store.begin(&self.chain_tip);
     }
     pub fn rollback(&mut self) {
-        self.marf.drop_current();
+        self.marf.as_mut().unwrap().drop_current();
         self.side_store.rollback(&self.chain_tip);
         self.chain_tip = TrieFileStorage::block_sentinel();
     }
     pub fn commit(&mut self) {
+        debug!("_metadata_ COMMIT()"); 
+        // AARON: I'm not sure this path should be considered 'legal' anymore,
+        //     and may want to delete or panic.
```

Comment:
Can you elaborate more on this comment?  Is it because both miners and peers have to call `commit_for_move` anyway, since the miner's call to `MARF::begin` will have had to have used a sentinel block hash?

Reply:
Exactly -- the only paths that use this now are in the test cases, I can remove.

Reply:
Addressed in aa9ff98